README.md

```diff
@@ -1,2 +1,116 @@
+
+This repository contains a [Gazelle plugin] used to generate [rules_swift] targets based upon your
+Swift source code..
+
+## Table of Contents
+
+<!-- MARKDOWN TOC: BEGIN -->
+* [Quickstart](#quickstart)
+  * [1. Enable bzlmod](#1-enable-bzlmod)
+  * [2. Configure your `MODULE.bazel` to use swift_gazelle_plugin.](#2-configure-your-modulebazel-to-use-swift_gazelle_plugin)
+  * [3. Add Gazelle targets to `BUILD.bazel` at the root of your workspace.](#3-add-gazelle-targets-to-buildbazel-at-the-root-of-your-workspace)
+  * [4. Create or update Bazel build files for your project.](#4-create-or-update-bazel-build-files-for-your-project)
+  * [5. Build and test your project.](#5-build-and-test-your-project)
+  * [6. Check in `MODULE.bazel`.](#6-check-in-modulebazel)
+  * [10. Start coding](#10-start-coding)
+* [Tips and Tricks](#tips-and-tricks)
+<!-- MARKDOWN TOC: END -->
+
+## Quickstart
+
+The following provides a quick introduction on how to set up and use the features in this
+repository. These instructions assume that you are using [Bazel modules] to load your external
+dependencies. If you are using Bazel's legacy external dependency management, we recommend using
+[Bazel's hybrid mode], then follow the steps in this quickstart guide.
+
+### 1. Enable bzlmod
+
+This repository supports [bzlmod].
+
+```
+common --enable_bzlmod
+```
+
+### 2. Configure your `MODULE.bazel` to use [swift_gazelle_plugin].
+
+Add a dependency on `swift_gazelle_plugin`.
+
+<!-- BEGIN MODULE SNIPPET -->
+
+```python
+bazel_dep(name = "swift_gazelle_plugin", version = "0.0.0")
+```
+
+<!-- END MODULE SNIPPET -->
+
+### 3. Add Gazelle targets to `BUILD.bazel` at the root of your workspace.
+
+Add the following to the `BUILD.bazel` file at the root of your workspace.
+
+```bzl
+load("@gazelle//:def.bzl", "gazelle", "gazelle_binary")
+
+# This declaration builds a Gazelle binary that incorporates all of the Gazelle
+# plugins for the languages that you use in your workspace. In this example, we
+# are only listing the Gazelle plugin for Swift from swift_gazelle_plugin.
+gazelle_binary(
+    name = "gazelle_bin",
+    languages = [
+        "@swift_gazelle_plugin//gazelle",
+    ],
+)
+
+# This target updates the Bazel build files for your project. Run this target
+# whenever you add or remove source files from your project.
+gazelle(
+    name = "update_build_files",
+    gazelle = ":gazelle_bin",
+)
+```
+
+### 4. Create or update Bazel build files for your project.
+
+Generate/update the Bazel build files for your project by running the following:
+
+```sh
+bazel run //:update_build_files
+```
+
+### 5. Build and test your project.
+
+Build and test your project.
+
+```sh
+bazel test //...
+```
+
+### 6. Check in `MODULE.bazel`.
+
+- The `MODULE.bazel` contains the declarations for your external dependencies.
```

Comment:
Should we mention MODULE.bazel.lock?

Reply:
Whether a workspace wants to check in their lock file is outside the scope of what we are trying to convey. There may be reasons for them not to do it. 🤷‍♂️